astropy/io/fits/file.py

```diff
@@ -49,6 +50,9 @@
     'wb': 'ostream', 'wb+': 'update',
     'ab': 'ostream', 'ab+': 'append'}
 
+# A match indicates the file was opened in text mode, which is not allowed
+TEXT_RE = re.compile('^[rwa]((t?\+?)|(\+?t?))$')
```

Comment:
There is a conflict with `master` now. In regards to squashing, IMHO it is best to do it as a last step when ready to merge; This is because reviewer sometimes just want to look at the last commit(s) since review, so squashing would cause unnecessary re-review.